### PR TITLE
Prevented bad eyes shader and badbone effects from being applied when in creative/spetator mode.

### PR DESCRIPTION
**Pull request by KiriCattus**
Though I'd suggest adding player capabilities for when the player moves from survival mode to creative/spetator mode if you wish to save pre-creative/spetator mode effects and duration for when the player returns to survival (I'm not that skilled enough to be able to do that, sorry :s)

### CODE REVIEW

File: src/main/java/subaraki/badbone/events/PlayerUpdateEvent.java [hunk 1]
```diff
@@ -130,4 +131,8 @@ private static void playHurtSound(Player player) {
         player.playNotifySound(SoundEvents.BONE_BLOCK_BREAK, SoundSource.PLAYERS, 1f, player.getRandom().nextFloat() + 1f);
         player.playNotifySound(SoundEvents.PLAYER_HURT, SoundSource.PLAYERS, 1f, player.getRandom().nextFloat() + 1f);
     }
+
+    public static boolean isInSurvivalMode(Player player) {
+        return !player.isSpectator() && !player.isCreative();
```
Comment (strikerrocker) [inline src/main/java/subaraki/badbone/events/PlayerUpdateEvent.java:136]: Just check for survival alone ? `player.isSurvival()`

File: src/main/java/subaraki/badbone/events/PlayerUpdateEvent.java [hunk 2]
```diff
@@ -104,7 +105,7 @@ public static void playerUpdate(TickEvent.PlayerTickEvent event) {
     @SubscribeEvent
     public static void playerFall(LivingHurtEvent event) {
         if (event.getEntityLiving() instanceof Player player) {
-            if (player.fallDistance > 3 && event.getSource().equals(DamageSource.FALL) && player.getRandom().nextInt(ConfigData.frequencyKnee) == 0) {
+            if (player.fallDistance > 1 && event.getSource().equals(DamageSource.FALL) && player.getRandom().nextInt(ConfigData.frequencyKnee) == 0  && isInSurvivalMode(player)) {
```
Comment (strikerrocker) [inline src/main/java/subaraki/badbone/events/PlayerUpdateEvent.java:108]: Why did you change the fall distance from 3 to 1?

Comment (KiriCattus) [reply src/main/java/subaraki/badbone/events/PlayerUpdateEvent.java]: Fixed now, was while I was testing to make sure things still worked :)

File: src/main/java/subaraki/badbone/events/PlayerUpdateEvent.java (hunk 1 above)
Comment (KiriCattus) [reply src/main/java/subaraki/badbone/events/PlayerUpdateEvent.java:136]: When I was looking around last night I couldn't for the life of me find an ``isSurvival()`` in the class where ``isCreative()``/``isSpetator()`` was located hence my making that method, why vanilla do this I don't know but this was the closest I could get.

**Review by ArtixAllMighty — APPROVED** [review]
Looks good. 
the creative check could be at the top, together with the server player check, but i'll edit that myself. 

thanks for the pr and testing !
